libnmstate/netapplier.py

```diff
@@ -333,3 +334,14 @@ def _cleanup_iface_ethernet_state_sanitize(desired_state, current_state):
             desired_state.pop('ethernet', None)
             current_state.pop('ethernet', None)
     return desired_state, current_state
+
+
+def _cleanup_iface_ipv6_link_local_addr(desired_state, current_state):
+    for state in (desired_state, current_state):
+        for address in state.get('ipv6', {}).get('address', []):
+            addr_str = '{a[ip]}/{a[prefix-length]}'.format(a=address)
+            if nm.ipv6.is_link_local_addr(addr_str):
+                state['ipv6']['address'].remove(address)
+        # Don't compare ipv6 status as link-local address might been removed.
+        if 'ipv6' in state and 'enabled' in state['ipv6']:
```

Comment:
I do not understand this one... we do not compare ipv6 at all? So what is the cleanup all about?

Reply:
We do not compare the `state['ipv6']['state']` entry, we still compare the `state['ipv6']['address']`.

Reply:
Oh, then if the user mentioned `disabled` and it is enabled with only the local-link, you want to pass verification?
That seems to hide reality: nmstate is unable to disable IPv6, why not express it?

BTW, a side question: What happens if we disable IPv6 using /proc/.. and then create a connection profile without the IPv6 setting? Will that still re-enable IPv6?

Reply:
NetworkManager will enable IPv6 when a profile got activated no match what previous proc state of that device is.

I don't think it's time to discuss that detail, we simple raise error and document that is not supported. We can come back to  that after we get a way to disable IPv6(via NetworkManager or disable IPv6 after NM profile activation).